src/test/java/com/app/gamaacademy/cabrasdoagrest/bankline/test/service/UsuarioServiceTest.java

```diff
@@ -0,0 +1,146 @@
+            service.criarUsuario(mapper.map(usuario, UsuarioDTO.class));
+        });
+
+        String mensagemEsperada = "O valor para [cpf] não é válido!";
+        String mansagemRecebida = exception.getMessage();
+
+        assertTrue(mansagemRecebida.contains(mensagemEsperada));
+	}
+
+    @Test
+	@Order(3)
+	@DisplayName("Deve lançar uma exceção ao tentar criar um usuário com senha inválida")
+	public void criandoUsuarioComSenhaInvalida() {
+		Usuario usuario = umUsuario.comSenhaInvalida().build();
+
+        Throwable exception = assertThrows(InvalidDataAccessApiUsageException.class, () -> {
+            service.criarUsuario(mapper.map(usuario, UsuarioDTO.class));
+        });
+
+        String mensagemEsperada = "O valor para [senha] não é válido!";
+        String mansagemRecebida = exception.getMessage();
+
+        assertTrue(mansagemRecebida.contains(mensagemEsperada));
+	}
+
+    @Test
+	@Order(4)
+	@DisplayName("Deve lançar uma exceção ao tentar criar um usuário com nome inválido")
+	public void criandoUsuarioComNomeInvalida() {
+		Usuario usuario = umUsuario.comNomeInvalido().build();
+
+        Throwable exception = assertThrows(InvalidDataAccessApiUsageException.class, () -> {
+            service.criarUsuario(mapper.map(usuario, UsuarioDTO.class));
+        });
+
+        String mensagemEsperada = "O valor para [nome] não é válido!";
+        String mansagemRecebida = exception.getMessage();
+
+        assertTrue(mansagemRecebida.contains(mensagemEsperada));
+	}
+
+    @Test
+	@Order(5)
+	@DisplayName("Deve lançar uma exceção ao tentar criar um usuário com login e/ou CPF já cadastrado")
+	public void criandoUsuarioComLoginExistente() {
+		Usuario usuario = umUsuario.build();
+        UsuarioDTO usuarioDTO = mapper.map(usuario, UsuarioDTO.class);
+        UsuarioServiceImpl usuarioService = mock(UsuarioServiceImpl.class);
+
+        when(usuarioService.criarUsuario(usuarioDTO))
+        .thenThrow(new DataIntegrityViolationException("CPF e/ou login já existe não é possível cadastrar!"));
+
+        Throwable exception = assertThrows(DataIntegrityViolationException.class, () -> {
+            usuarioService.criarUsuario(usuarioDTO);
+        });
+
+        String mensagemEsperada = "CPF e/ou login já existe não é possível cadastrar!";
+        String mansagemRecebida = exception.getMessage();
+
+        assertTrue(mansagemRecebida.contains(mensagemEsperada));
+	}
+
+    @Test
+	@Order(6)
+	@DisplayName("Deve retornar um id ao criar um usuário")
+	public void criandoUsuario() {
+		Usuario usuario = umUsuario.comId().build();
+        UsuarioRepository usuarioRepository = mock(UsuarioRepository.class);
+			
+        when(usuarioRepository.save(usuario)).thenReturn(usuario);
+        service.setRepository(usuarioRepository);
+
+        Integer idEsperado = usuario.getId();
+        Integer idRecebido = service.criarUsuario(mapper.map(usuario, UsuarioDTO.class));
+
+        assertEquals(idEsperado, idRecebido);
+	}
```

Comment:
Retirar a inicialização do objeto de mock para um método anotado com @before.

usuarioRepository deve ser uma propriedade da classe com anotação @MockBean.

https://youtu.be/4VmbETu-dcA?t=1335